tests/Bootstrap.php

```diff
@@ -0,0 +1,243 @@
+<?php
+/**
+ * @author Todd Burry <todd@vanillaforums.com>
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+namespace VanillaTests;
+
+use Garden\Container\Container;
+use Garden\Container\Reference;
+use Gdn;
+use Interop\Container\ContainerInterface;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+use Vanilla\Addon;
+use Vanilla\InjectableInterface;
+
+/**
+ * Run bootstrap code for Vanilla tests.
+ *
+ * This class is meant to be re-used. Calling {@link Bootstrap::run()} on a polluted environment should reset it.
+ */
+class Bootstrap {
+    private $baseUrl;
+
+    /**
+     * Bootstrap constructor.
+     *
+     * A different base URL affects
+     *
+     * @param string $baseUrl The base URL of the installation.
+     */
+    public function __construct($baseUrl = 'http://vanilla.test') {
```

Comment:
That's something that can be added later. Since these internal tests don't actually run against a site though it only affects  URLs.